### ISSUE TITLE
Add an analyzer to suggest always await task when there is any using statements in the local context

### ISSUE THREAD

**Issue by Cosifne** [dotnet/roslyn-analyzers#6074]
Hi, recently I fell into a pitfall which makes me feel it would be a good idea to have an analyzer to detect this.

Consider the following code:
```
public Task Hello()
{
    using var bar = new Bar();
    return DoSomethingAsync();
}
```

```
public async Task Hello()
{
    using var bar = new Bar();
    return await DoSomethingAsync();
}
```
The difference is:
In the first example, `bar` is disposed after when returned from `Hello`. (`DoSomethingAsync` might not complete at that time)
In the second example, `bar is disposed after `DoSomethingAsync` is completed. 

This behavior of 'async' method might lead to strange behavior and it's very trivia to debug.

I suggest we should give a suggestion diagnostics at the return statement if
1. The return statement is leaving from a scope, which contains 'using' statement/expressions
2. The method is not async, and the return statement's type is available.

The code fix would be converting the return statement to 'await' pattern, and the containing method to 'async' version.

**Comment by CollinAlpert** [comment]
I like this a lot. The same warning should be reported inside a try-block. Tasks not awaited inside a try-block might not be caught by the corresponding catch-block.
